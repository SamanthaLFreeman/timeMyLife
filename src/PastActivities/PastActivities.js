import React from 'react';
import Activity from '../Activity/Activity';

const PastActivities = ({activities}) => {
  const displayActivities = activities.map(activity => <Activity activity={activity} key={activity.name+activity.category} />);

  return (
    <section className='PastActivities'>
      <h2>Past Activities</h2>
      {displayActivities}
    </section>
  )
}

export default PastActivities;