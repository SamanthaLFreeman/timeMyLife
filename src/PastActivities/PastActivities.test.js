import React from 'react';
import PastActivities from './PastActivities';
import { shallow } from 'enzyme';

describe('PastActivities', () => {
  let wrapper;
  const activitiesMock = [{
    category: 'Study',
    name: 'Activity name',
    mins: '1',
    secs: '30'
  }]

  beforeEach(() => {
    wrapper = shallow(
    <PastActivities 
      activities={activitiesMock}/>)
  })

  it('should match the snapshot with the correct data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })
})