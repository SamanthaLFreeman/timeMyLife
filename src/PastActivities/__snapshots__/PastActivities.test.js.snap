// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastActivities should match the snapshot with the correct data passed through 1`] = `
<section
  className="PastActivities"
>
  <h2>
    Past Activities
  </h2>
  <Activity
    activity={
      Object {
        "category": "Study",
        "mins": "1",
        "name": "Activity name",
        "secs": "30",
      }
    }
    key="Activity nameStudy"
  />
</section>
`;
